reconcile/utils/saasherder.py

```diff
@@ -1441,6 +1505,93 @@ def get_upstream_jobs_diff_saas_file(
 
         return trigger_specs
 
+    def get_container_images_diff(
+        self, dry_run: bool
+    ) -> list[TriggerSpecContainerImage]:
+        results = threaded.run(
+            self.get_container_images_diff_saas_file,
+            self.saas_files,
+            self.thread_pool_size,
+            dry_run=dry_run,
+        )
+        return list(itertools.chain.from_iterable(results))
+
+    def get_container_images_diff_saas_file(
```

Comment:
Can you write some documentation explaining what this function does?

Reply:
https://github.com/app-sre/qontract-reconcile/pull/2833/commits/403ccec8f78666fc20665c269d390212cb4218c2